doc/sourmash-internals.md

```diff
@@ -0,0 +1,648 @@
+
+In practice, sourmash does all necessary downsampling dynamically, but
+returns the original sketches. This means that (for example) you can
+do a low-resolution/high-scaled large scale search by specifying a
+high scaled value, and then do a higher resolution comparison with
+only the highly similar matches the results to do a more refined (see
+below, Speeding up `gather` and `search`.)
+
+### Num (MinHash) sketches support Jaccard similarity
+
+"Regular" MinHash (or "num MinHash") sketches are implemented the same
+way as in mash.  However, they are less well supported in sourmash,
+because they don't offer the same opportunities for metagenome
+analysis.  (See also [sourmash#1354](https://github.com/sourmash-bio/sourmash/issues/1354).)
+
+Num MinHash sketches can always be compared by downsampling to a
+common `num` value. This may need to be done manually using `sourmash
+sig downsample`, however.
+
+## K-mer sizes
+
+There is no explicit restriction on k-mer sizes built into sourmash.
+
+For highly specific genome and metagenome comparisons, we typically
+use k=21, k=31, or k=51. For a longer discussion, see [Assembly Free Analysis with K-mers](https://github.com/mblstamps/stamps2022/blob/main/kmers_and_sourmash/2022-stamps-assembly-free%20class.pdf) from STAMPS 2022.
+
+## Molecule types - DNA, protein, Dayhoff, and hydrophobic-polar
+
+sourmash supports four different sequence encodings, which we refer to
+as "molecule: DNA (`--dna`), protein (`--protein`), Dayhoff,
+(`--dayhoff`), and hydrophobic-polar (`--hp`).
+
+All FracMinHash sketches have exactly one molecule type, and can only
+be compared to the same molecule type (and ksize).
+
+DNA moltype sketches can be constructed from DNA input sequences using
+`sourmash sketch dna`.
+
+Protein, Dayhoff, and HP moltype sketches can be constructed from
+protein input sequences using `sourmash sketch protein`, or from DNA
+input sequences using `sourmash sketch translate`; `translate` will
+translate in all six reading frames (see also
+[orpheum](https://github.com/czbiohub/orpheum) from
+[Botvinnik et al., 2021](https://www.biorxiv.org/content/10.1101/2021.07.09.450799v1)).
+By default protein sketches will be created; dayhoff sketches can be
+created by including `dayhoff` in the param string, e.g. `sourmash
+sketch protein -p dayhoff`, and hydrophobic-polar sketches can be
+built with `hp` in the param string.
+
+## Manifests
+
+sourmash makes extensive use of signature manifests to support rapid
+selection and lazy loading of signatures based on signature metadata
+(name, ksize, moltype, etc.) See
+[Blog post: Scaling sourmash to millions of samples](http://ivory.idyll.org/blog/2021-sourmash-scaling-to-millions.html)
+for some of the motivation.
+
+Manifests are an internal format that is not meant to be particularly
+human readable, but the CSV format can be loaded into a spreadsheet
+program if you're curious :).
+
+If in a zipfile (`.zip`) or SBT zipfile (`.sbt.zip`), manifests must
```

Comment:
```suggestion
If a collection of signatures are in a zipfile (`.zip`) or SBT zipfile (`.sbt.zip`), manifests must
```

Reply:
thanks! are => is, doing this in separate commit.

Reply:
updated in fbbcee56